### PR TITLE
[GAL-3844] add Close Confirmation warning to Post Composer

### PR DESCRIPTION
**Pull request by Rohan-cp**
Screenshot 2023-08-02 at 5 02 45 PM

### CODE REVIEW

File: apps/web/src/components/Posts/DiscardPostConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    hideModal();
+    onDiscard();
+  };
+
+  return (
+    <StyledConfirmation>
+      <StyledTextWrapper>
+        <BaseM>Are you sure?</BaseM>
+      </StyledTextWrapper>
+      <BaseM>If you go back now, this post will be discarded</BaseM>
```
Comment (kaitoo1) [inline apps/web/src/components/Posts/DiscardPostConfirmation.tsx:26]:
```suggestion
      <BaseM>If you go back now, this post will be discarded.</BaseM>
```

**Review by kaitoo1 — APPROVED** [review]
This works great!! thanks Rohan


would you also be able to show the Discard Post confirmation when the user clicks the back button in the composer? 🙏 🙏 

otherwise this is awesome. love that it shows the confirmation when you click on close or click outside of the modal as well. 

CleanShot 2023-08-02 at 21 43 34

File: apps/web/src/components/Posts/DiscardPostConfirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  onDiscard: () => void;
+};
+
+export default function DiscardPostConfirmation({ onDiscard }: Props) {
+  const { hideModal } = useModalActions();
+
+  const handleDiscardConfirmClick = () => {
+    hideModal();
+    onDiscard();
+  };
```
Comment (Robinnnnn) [inline apps/web/src/components/Posts/DiscardPostConfirmation.tsx:19]: let's wrap in `useCallback` 🙏

File: apps/web/src/contexts/modal/AnimatedModal.tsx [hunk 3]
```diff
@@ -81,9 +83,17 @@ function AnimatedModal({
     return 'unset';
   }, [isFullPage, isMobile]);
 
+  const onClickHandler = () => {
+    if (onBypassClose) {
+      onBypassClose();
+    } else {
+      hideModal();
+    }
+  };
```
Comment (Robinnnnn) [inline apps/web/src/contexts/modal/AnimatedModal.tsx:92]: 1. wrap in `useCallback`
2. could we rename this function from `onClickHandler` -> `handleClick`

File: apps/web/src/contexts/modal/AnimatedModal.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ type Props = {
   headerText: string;
   headerVariant: ModalPaddingVariant;
   hideClose?: boolean;
+  onBypassClose?: () => void;
```
Comment (Robinnnnn) [inline apps/web/src/contexts/modal/AnimatedModal.tsx:34]: could we also rename this to `onCloseOverride`, sorry

File: apps/web/src/contexts/modal/ModalContext.tsx [hunk 5]
```diff
@@ -45,6 +45,7 @@ type ShowModalFnProps = {
   hideClose?: boolean;
   isPaddingDisabled?: boolean;
   onClose?: () => void;
+  onBypassClose?: () => void;
```
Comment (Robinnnnn) [inline apps/web/src/contexts/modal/ModalContext.tsx:48]: could we add a comment above this line explaining what this prop does?

e.g. `// if provided, this callback will trigger as opposed to closing the modal`

**Pull request merged**